### DISCUSSION TITLE
Custom bulk import/update of products by SKU

### DISCUSSION THREAD

**Discussion by thardyman** [vendurehq/vendure#1333] [Q&A]
Hi, we have a reasonably large inventory, 3000+ items, which we have successfully loaded using the Populate function.  

Moving forward we would like to be able to create / update these products periodically.  We will receive a csv file with updated prices / descriptions.  Our business works with a "Product Code" to identify unique items that we sell.  I have chosen to map this to the `productVariant.sku`  field.

I'm now in the process of writing a custom plugin that will process a csv file which contains all the updates.  It might also include new products.  Since I don't know the corresponding Vendure variant's product id, I need to try and find variants by their SKU. This is where I'm stuck.

I might be approaching this all wrong. Any ideas?

**Comment by michaelbromley** [comment]
Hi!
This sounds really similar to my own project, where I also map product code -> sku and update based on that. So the general idea is sound. What exactly are you stuck on?

**Reply by thardyman** [reply]
Cool, that's good to know.  It's the bit where I want to select a product variant by its SKU.  I can't work out how to do it. I see the methods `findAll` `findOne` `findByIds` `getVariantsByProductId`  on the `ProductVariantService` not sure I want any of those. I really need a method to search by attribute.

Thanks!

**Reply by michaelbromley** [reply]
In cases like this, where the built-in services do not cover your needs, you should use the TypeORM APIs directly. For instance:

```ts
const variant = await this.connection.getRepository(ctx, ProductVariant).findOne({ where: { sku: productCode } });
```

**Reply by thardyman** [reply]
Perfect, thanks! That's really helped.
